src/Traits/Activatable.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodeIgniter\Shield\Traits;
+
+trait Activatable
+{
+    /**
+     * Returns true if the user has been activated
+     * and activation is required after registration.
+     */
+    public function isActivated(): bool
+    {
+        // If activation is not required, then we're always active.
+        return ! $this->shouldActivate() || $this->active;
+    }
+
+    /**
+     * Returns true if the user has not been activated.
+     */
+    public function isNotActivated(): bool
+    {
+        return ! $this->isActivated();
+    }
+
+    /**
+     * Activates the user.
+     */
+    public function activate(): bool
```

Comment:
Can you change the return type to `void`?
The `UserModel::update()` throws an exception when the update fails.
So returning `bool` does not make sense.

Reply:
Unless it's changing in a PR, `update()` currently [returns a boolean](https://github.com/codeigniter4/shield/blob/develop/src/Models/UserModel.php#L268). Changing this to `void` throws errors.

Reply:
I sent a PR #626